taskcluster/scriptworker_taskgraph/transforms/python_version.py

```diff
@@ -0,0 +1,91 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Create a task per job python-version
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+from copy import deepcopy
+
+from taskgraph.transforms.base import TransformSequence
+
+
+transforms = TransformSequence()
+
+def _replace_string(obj, subs):
+    if isinstance(obj, dict):
+        return {k: v.format(**subs) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        for c in range(0, len(obj)):
+            obj[c] = obj[c].format(**subs)
+    else:
+        obj = obj.format(**subs)
+    return obj
+
+
+def _resolve_replace_string(item, field, subs):
+    # largely from resolve_keyed_by
+    container, subfield = item, field
+    while '.' in subfield:
+        f, subfield = subfield.split('.', 1)
+        if f not in container:
+            return item
+        container = container[f]
+        if not isinstance(container, dict):
+            return item
+
+    if subfield not in container:
+        return item
+
+    container[subfield] = _replace_string(container[subfield], subs)
+    return item
+
+
+
+@transforms.add
+def set_script_name(config, jobs):
+    for job in jobs:
+        job.setdefault("attributes", {}).update({
+            "script-name": job["name"],
+        })
+        yield job
+
+
+@transforms.add
+def tasks_per_python_version(config, jobs):
+    fields = [
+        "description",
+        "docker-repo",
+        "run.command",
+        "worker.command",
+        "worker.docker-image",
+    ]
+    for job in jobs:
+        for python_version in job.pop("python-versions"):
+            task = deepcopy(job)
+            subs = {"name": job["name"], "python_version": python_version}
+            for field in fields:
+                _resolve_replace_string(task, field, subs)
+            task["attributes"]["python-version"] = python_version
+            yield task
```

Comment:
Followup: I wonder if the python version logic shouldn't be part of the dedicated loader. In android-components we also have 2 dimensions: 

1. Subprojects
2. Build types (regular, nightly, release)

The specific loader handles [the latter there](https://github.com/mozilla-mobile/android-components/blob/6744e62723898db1c5beb24f926303d9279b6030/taskcluster/ac_taskgraph/loader/build_config.py#L29).

Reply:
Could do. [If we add a non-python script](https://github.com/mozilla-releng/scriptworker-scripts/pull/211), we either have to adjust the transforms or the loader, depending on which solution we go with. Which would make more sense?

Reply:
It is not clear to me that the python-version transform is providing value for the `k8s-image` kind in any case.

Reply:
I could see us creating k8s images for multiple python versions at some point, especially after we drop support for docker hub. I would like to land this PR at some point sooner rather than later, as well.

Reply:
Yeah, I definitely wasn't suggesting changing anything here, just commenting on the future consideration with non-python things